src/isomorphic/classic/element/ReactElement.js

```diff
@@ -264,6 +264,16 @@ ReactElement.cloneElement = function(element, config, children) {
     props.children = childArray;
   }
 
+  // Resolve default props
+  if (element.type && element.type.defaultProps) {
+    var defaultProps = element.type.defaultProps;
+    for (propName in defaultProps) {
```

Comment:
We probably don't need to introduce a loop here, since we know the only way this could happen is if the config prop is undefined.  We can just add an undefined check to the existing loop above.